sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -444,12 +444,42 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
             write.query.schema.foreach(f => TypeUtils.failWithIntervalType(f.dataType))
 
           case alter: AlterTableCommand if alter.table.resolved =>
+            val table = alter.table.asInstanceOf[ResolvedTable]
+            def findField(fieldName: Seq[String]): StructField = {
+              // Include collections because structs nested in maps and arrays may be altered.
+              val field = table.schema.findNestedField(fieldName, includeCollections = true)
+              if (field.isEmpty) {
+                alter.failAnalysis(s"Cannot ${alter.operation} missing field ${fieldName.quoted} " +
+                  s"in ${table.name} schema: ${table.schema.treeString}")
+              }
+              field.get._2
+            }
+            def findParentStruct(fieldNames: Seq[String]): StructType = {
```

Comment:
`findField` and `findParentStruct` are existing functions in `CheckAnalysis` with a slight modification (e.g., not passing operation argument). The existing functions will be removed once all alter table commands are migrated.